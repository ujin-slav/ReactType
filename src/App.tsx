import React from 'react';
import Form from './components/Form';
import UserList from './components/UserList';

export interface IAuto {
  name: string,
  engine: string,
  power:number
}

function App() {

  const lustAuto:IAuto[] = [
    {
      name: "Volvo",
      engine: "1.6",
      power:100
    },
    {
      name: "VW",
      engine: "2.0",
      power:200
    },
    {
      name: "Opel",
      engine: "2.6",
      power:300
    }
  ]

  return (
    <div >
      {/* <UserList/> */}
      <Form listAuto={lustAuto}/>
    </div>
  );
}

export default App;
