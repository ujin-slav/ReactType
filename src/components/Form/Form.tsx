import React, { FC, useEffect, useState } from 'react';
import { IAuto } from '../../App';
import Search from '../Search/Search';

type Props = {
    listAuto ?:IAuto[]
}

const auto:IAuto[] = [
    {
      name: "Volvo",
      engine: "1.6",
      power:100
    },
    {
      name: "VW",
      engine: "2.0",
      power:200
    },
    {
      name: "Opel",
      engine: "2.6",
      power:300
    }
  ]

const Form:FC<Props> = ({...props}) => {

    //const{listAuto = []} = props 
    const [searchText,setSearchText] = useState<string>('')
    const [listAuto,setListAuto] = useState<IAuto[]>(auto)

    // if(listAuto.length===0){
    //     return null
    // }

    const handlerSearch = (e:React.ChangeEvent<HTMLInputElement>):void =>{ 
        const text = e.target.value
        setListAuto(auto.filter((item)=>item.name.includes(text)))
    }

    return (
        <div>
            <Search onChange={handlerSearch}/>
            <ul>
                {listAuto.map((item, index)=>
                    <li key={index}>{item.name}</li>
                )}
            </ul>
        </div>
    );
};

export default Form;