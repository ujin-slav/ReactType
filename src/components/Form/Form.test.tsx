import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Form from './Form';
import { IAuto } from '../../App';

const lustAuto:IAuto[] = [
    {
      name: "Volvo",
      engine: "1.6",
      power:100
    },
    {
      name: "VW",
      engine: "2.0",
      power:200
    },
    {
      name: "Opel",
      engine: "2.6",
      power:300
    }
  ]

describe('Test component form',()=>{
    test('renders learn react link', () => {
        render(<Form listAuto={lustAuto}/>);
        const linkElement = screen.getByText(/Opel/i);
        expect(screen.getByRole('list')).toBeInTheDocument()
        expect(linkElement).toBeInTheDocument()
    });
    test('renders without data', () => {
      render(<Form/>);
      // const linkElement = screen.queryByRole('list');
      // expect(linkElement).not.toBeInTheDocument()
    });
    test('List snapshot', () => {
      const { container } = render(<Form listAuto={lustAuto}/>)
      expect(container).toMatchSnapshot()
    });
})

