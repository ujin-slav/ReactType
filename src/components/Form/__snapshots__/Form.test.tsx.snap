// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test component form List snapshot 1`] = `
<div>
  <div>
    <label>
      <input
        type="text"
        value=""
      />
    </label>
    <ul>
      <li>
        Volvo
      </li>
      <li>
        VW
      </li>
      <li>
        Opel
      </li>
    </ul>
  </div>
</div>
`;
